File: algorithm-analysis/homework-001/c-code/algorithmTests.c
/* Authors: Pedro Ribeiro, Gabriela Damazo & Isabelli Tedeschi
 * Description: Algorithm time tests
 * Date: 28/05/2022 - Hour: 11:41
 */

#include <stdlib.h>
#include <time.h>
#include "sortingAlgorithms.h"

double insertionSortTest(int *v, int n) {

    clock_t startTest, endTest;

    double timeSpent;

    startTest = clock();
    insertionSort(v, n);
    endTest = clock();

    timeSpent = ((double) endTest - startTest) / CLOCKS_PER_SEC;

    return timeSpent;

}

double modifiedInsertionSortTest(int *v, int left, int right) {

    clock_t startTest, endTest;

    double timeSpent;

    startTest = clock();
    modifiedInsertionSort(v, left, right);
    endTest = clock();

    timeSpent = ((double) endTest - startTest) / CLOCKS_PER_SEC;

    return timeSpent;

}

double mergeSortTest(int *v, int left, int right) {

    clock_t startTest, endTest;

    double timeSpent;

    startTest = clock();
    mergeSort(v, left, right);
    endTest = clock();

    timeSpent = ((double) endTest - startTest) / CLOCKS_PER_SEC;

    return timeSpent;

}

double modifiedMergeSortTest(int *v, int n, int left, int right) {

    clock_t startTest, endTest;

    double timeSpent;

    startTest = clock();
    modifiedMergeSort(v, n, left, right);
    endTest = clock();

    timeSpent = ((double) endTest - startTest) / CLOCKS_PER_SEC;

    return timeSpent;

}

double randomQuickSortTest(int *v, int left, int right) {

    clock_t startTest, endTest;

    double timeSpent;

    startTest = clock();
    randomQuickSort(v, left, right);
    endTest = clock();

    timeSpent = ((double) endTest - startTest) / CLOCKS_PER_SEC;

    return timeSpent;

}

double modifiedQuickSortTest(int *v, int n, int left, int right) {

    clock_t startTest, endTest;

    double timeSpent;

    startTest = clock();
    modifiedQuickSort(v, n, left, right);
    endTest = clock();

    timeSpent = ((double) endTest - startTest) / CLOCKS_PER_SEC;

    return timeSpent;

}

File: algorithm-analysis/homework-001/python-code/algorithmTests.py
# Authors: Pedro Ribeiro & Paulo Silva
# Description: Algorithm performance tests
# Date: 28/05/2022 - Hour: 11:14

import insertionSort as insertion
import mergeSort as merge
import quickSort as quick
import time

def insertionSortTest(v):

    startTest = time.time()
    insertion.insertionSort(v)
    endTest = time.time()
    spendTime = endTest - startTest

    return spendTime

def modifiedInsertionSortTest(v):

    n = len(v)

    startTest = time.time()
    insertion.modifiedInsertionSort(v, 0, n - 1)
    endTest = time.time()
    spendTime = endTest - startTest

    return spendTime

def mergeSortTest(v):

    n = len(v)
    aux = [0] * len(v)

    startTest = time.time()
    merge.mergeSort(v, aux, 0, n - 1)
    endTest = time.time()
    spendTime = endTest - startTest

    return spendTime

def modifiedMergeSortTest(v):

    n = len(v)
    aux = [0] * len(v)

    startTest = time.time()
    merge.modifiedMergeSort(v, aux, 0, n - 1)
    endTest = time.time()
    spendTime = endTest - startTest

    return spendTime

def quickSortTest(v):

    n = len(v)

    startTest = time.time()
    quick.randomQuickSort(v, 0, n - 1)
    endTest = time.time()
    spendTime = endTest - startTest

    return spendTime

def modifiedQuickSortTest(v):

    n = len(v)

    startTest = time.time()
    quick.modifiedQuickSort(v, 0, n - 1)
    endTest = time.time()
    spendTime = endTest - startTest

    return spendTime